airqo-platform/AirQo-api
# eliminate duplicate entries for days

# Hotfix

**_WHAT DOES THIS PR DO?_**

- [x] This partly addresses this [issue-273](https://github.com/airqo-platform/AirQo-api/issues/273). Attempts to resolve the second sub-issue about duplicate entries for days which were brought after more unit tests of a recently deployed cloud function.
- [x] Set ups the unit test files for all functions and also installs the relevant dependencies.
- [x] Refactoring to rename file names 

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-460](https://airqoteam.atlassian.net/browse/PLAT-460?atlOrigin=eyJpIjoiZThkNDllNGQzNjI0NGQyN2JlN2Q1ZjM0ZGM4Mjg2NmIiLCJwIjoiaiJ9)

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/hotfix-duplicate-days

**_HOW DO I TEST OUT THIS PR?_**
`cd AirQo-api/src/device-registry`
_Start your local redis server accordingly_
_Open a new tab and ensure you are still in the microservice directory(device-registry) and then run:_
`npm run dev-mac` for Linux and` npm run dev-pc `for Windows

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
BASE_URL: http://localhost:3000/api/v1/devices/
ADD EVENTS PARTIAL URL [POST]: events/add?device={DEVICE_NAME}&tenant={ORGANISATION_NAME}

BODY:
An array of one or more objects that represent device measurements for various sensors.
```
  [
    {
        "frequency": "minute",
        "time": "2020-10-29T06:38:56Z",
        "channelID": 233656
    },
    {
        "frequency": "minute",
        "time": "2020-03-16T13:04:17Z",
        "channelID": 233656
       
    },
    {
        "frequency": "minute",
        "time": "2020-10-28T17:34:16Z",
        "channelID": 233656
    }
]
```

To test out, find a way of making multiple inserts of the same measurements at the same time using different HTTP REST clients. And then observe if there are any repetitions (days) of values on the same days.
Keep on changing the **device name** in the POST URL and observe behaviour of insertion in the DB

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A
